tests/component/system/test_device_properties.py

```diff
@@ -112,3 +112,70 @@ def test___module___get_chassis___shared_interpreter(device: Device):
     chassis = device.compact_daq_chassis_device
 
     assert chassis._interpreter is device._interpreter
+
+
+def test___ext_cal_last_date_and_time___no_errors(any_x_series_device: Device) -> None:
+    last_date_and_time = any_x_series_device.ext_cal_last_date_and_time
+
+    assert last_date_and_time is not None
```

Comment:
At least validate we're getting the right type. I would also like to validate the data a little. A simulated devices seem to return time 0 with the 1904 epoch. A real device should return something sort of reasonable, like make sure the year is >2009? So maybe you create a test with a simulated_x_series and one with a real one? (same for selfcal)

Reply:
I get error -200077 (INVALID_ATTRIBUTE_VALUE) when getting `ext_cal_last_date_and_time` on simulated device, maybe the attribute only available for real device?

![image](https://github.com/ni/nidaqmx-python/assets/27721199/98a34872-9838-4af2-9493-13a4e2aee650)

Cal date on simulated device - `Dev1` on MAX:
![image](https://github.com/ni/nidaqmx-python/assets/27721199/d8397728-1499-4880-8b0c-fe4cbd7bc52e)

Reply:
Ah dang, yeah, it must be real-device only. Updating to `real_xseries_device` or whatever its called and doing a bit better validation sounds good

Reply:
Updated `ext_cal_last_date_and_time` and `self_cal_last_date_and_time` test cases with real x series device.